File: 0452-Minimum-Number-of-Arrows-to-Burst-Balloons.js
var findMinArrowShots = function(points) {
    points.sort((p1, p2) => p1[1] - p2[1]);

    let numArrows = 1;
    let currentEnd = points[0][1];

    for (let i = 1; i < points.length; i++) {
        if (points[i][0] > currentEnd) {
            numArrows ++;
            currentEnd = points[i][1];
        }
    }

    return numArrows;
};

File: 0452-Minimum-Number-of-Arrows-to-Burst-Balloons.py
class Solution(object):
    def findMinArrowShots(self, points):
        points.sort(reverse = True)
        currentArrow = points[0][0]
        arrowCount = 1

        for leftEnd, rightEnd in points:
            if currentArrow > rightEnd:
                currentArrow = leftEnd
                arrowCount += 1

        return arrowCount
